exercises/forth/canonical-data.json

```diff
@@ -311,6 +311,53 @@
           "expected": null
         }
       ]
+    },
+    {
+      "description": "case-insensitivity",
+      "cases": [
+        {
+          "description": "DUP is case-insensitive",
+          "property": "evaluate",
+          "input": ["1 DUP Dup dup"],
+          "expected": [1, 1, 1, 1]
+        },
+        {
+          "description": "DROP is case-insensitive",
+          "property": "evaluate",
+          "input": ["1 2 3 4 DROP Drop drop"],
+          "expected": [1]
+        },
+        {
+          "description": "SWAP is case-insensitive",
+          "property": "evaluate",
+          "input": ["1 2 SWAP 3 Swap 4 swap"],
+          "expected": [2, 3, 4, 1]
+        },
+        {
+          "description": "OVER is case-insensitive",
+          "property": "evaluate",
+          "input": ["1 2 OVER Over over"],
+          "expected": [1, 2, 1, 2, 1]
+        },
+        {
+          "description": "user-defined words are case-insensitive",
+          "property": "evaluate",
+          "input": [
+            ": foo dup ;",
+            "1 FOO Foo foo"
+          ],
+          "expected": [1, 1, 1, 1]
+        },
+        {
+          "description": "overridden words are case-insensitive",
+          "property": "evaluate",
+          "input": [
+            ": swap dup ;",
+            "1 SWAP Swap swap"
```

Comment:
I find it unlikely that any implementation will fail this test if it's the case that they check that user-defined words are case-insensitive, since at this point swap is a user-defined word, but I'm happy to take your word for it.

I would consider something that checks that definition bodies are case-insensitive, I suppose? `: swap DUP Dup dup ;`, etc.

Reply:
@petertseng, that's a good point - would you be happy with the below alternative? I could envisage an implementation treating definitions separately and forgetting to normalise the case, and I think that this should catch any such oversight.

```
: SWAP DUP Dup dup ;
1 swap
```

Reply:
👍